contracts/core/DaoRegistry.sol

```diff
@@ -219,6 +221,52 @@ contract DaoRegistry is MemberGuard, AdapterGuard {
         emit ConfigurationUpdated(key, value);
     }
 
+    function jailMember(address memberAddress)
+        external
+        hasAccess(this, AclFlag.JAIL_MEMBER)
+    {
+        require(memberAddress != address(0x0), "invalid member address");
+
+        Member storage member = members[memberAddress];
+        require(
+            DaoHelper.getFlag(member.flags, uint8(MemberFlag.EXISTS)),
+            "member does not exist"
+        );
+
+        member.flags = DaoHelper.setFlag(
+            member.flags,
+            uint8(MemberFlag.JAILED),
+            true
+        );
+    }
+
+    function notJailed(address memberAddress) external view returns (bool) {
+        return
+            !DaoHelper.getFlag(
+                members[memberAddress].flags,
+                uint8(MemberFlag.JAILED)
+            );
+    }
+
+    function unjailMember(address memberAddress)
+        external
+        hasAccess(this, AclFlag.JAIL_MEMBER)
+    {
+        require(memberAddress != address(0x0), "invalid member address");
+
+        Member storage member = members[memberAddress];
+        require(
```

Comment:
should we add another check to verify if the member is in jail?

Reply:
same as with jailMember, if the member is not jailed it just won't do anything